sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java

```diff
@@ -74,44 +78,90 @@
 @Singleton
 public class AlterTableOperation {
 
+    private static final String UNNAMED = SQLOperations.Session.UNNAMED;
     private static final Function<Object, Long> LONG_NULL_FUNCTION = Functions.constant(null);
     private final ClusterService clusterService;
     private final TransportActionProvider transportActionProvider;
+    private final SQLOperations sqlOperations;
 
     @Inject
-    public AlterTableOperation(ClusterService clusterService, TransportActionProvider transportActionProvider) {
+    public AlterTableOperation(ClusterService clusterService,
+                               TransportActionProvider transportActionProvider,
+                               SQLOperations sqlOperations) {
         this.clusterService = clusterService;
         this.transportActionProvider = transportActionProvider;
+        this.sqlOperations = sqlOperations;
     }
 
     public ListenableFuture<Long> executeAlterTableAddColumn(final AddColumnAnalyzedStatement analysis) {
         final SettableFuture<Long> result = SettableFuture.create();
         if (analysis.newPrimaryKeys() || analysis.hasNewGeneratedColumns()) {
-            String stmt = String.format(Locale.ENGLISH, "SELECT COUNT(*) FROM \"%s\".\"%s\"", analysis.table().ident().schema(), analysis.table().ident().name());
-            transportActionProvider.transportSQLAction().execute(new SQLRequest(stmt), new ActionListener<SQLResponse>() {
-                @Override
-                public void onResponse(SQLResponse sqlResponse) {
-                    Long count = (Long) sqlResponse.rows()[0][0];
-                    if (count == 0L) {
-                        addColumnToTable(analysis, result);
-                    } else {
-                        String columnFailure = analysis.newPrimaryKeys() ? "primary key" : "generated";
-                        result.setException(new UnsupportedOperationException(String.format(Locale.ENGLISH,
-                            "Cannot add a %s column to a table that isn't empty", columnFailure)));
-                    }
-                }
+            String schema = analysis.table().ident().schema();
```

Comment:
Could also assign the ident to a local variable to reduce the redundant function calls.